tests/utils/tensorflow/test_rasa_layers.py

```diff
@@ -0,0 +1,748 @@
+    "attribute_signature, inputs, expected_outputs",
+    [
+        # Both sequence- and sentence-level features
+        (
+            {
+                SEQUENCE: [
+                    realistic_feature_signature_dense_1,
+                    realistic_feature_signature_dense_2,
+                ],
+                SENTENCE: [realistic_feature_signature_dense_3,],
+            },
+            (
+                [realistic_feature_dense_seq_1, realistic_feature_dense_seq_2],
+                [realistic_feature_dense_sent_3],
+                realistic_sequence_lengths,
+            ),
+            (
+                np.array(
+                    [
+                        [
+                            [10.0, 1.0, 2.0],
+                            [20.0, 3.0, 4.0],
+                            [30.0, 5.0, 6.0],
+                            [0.1, 0.2, 0.3],
+                        ],
+                        [
+                            [40.0, 1.5, 2.5],
+                            [50.0, 3.5, 4.5],
+                            [0.4, 0.5, 0.6],
+                            [0.0, 0.0, 0.0],
+                        ],
+                    ],
+                    dtype=np.float32,
+                ),
+                [[[1.0], [1.0], [1.0], [1.0]], [[1.0], [1.0], [1.0], [0.0]]],
+                np.concatenate(
+                    (realistic_feature_dense_seq_1, [[[0.0]], [[0.0]]]), axis=1
+                ),
+            ),
+        ),
+        # Only sequence-level features
+        (
+            {
+                SEQUENCE: [
+                    realistic_feature_signature_dense_1,
+                    realistic_feature_signature_dense_2,
+                ],
+                SENTENCE: [],
+            },
+            (
+                [realistic_feature_dense_seq_1, realistic_feature_dense_seq_2],
+                [],
+                realistic_sequence_lengths,
+            ),
+            (
+                np.array(
+                    [
+                        [[10.0, 1.0, 2.0], [20.0, 3.0, 4.0], [30.0, 5.0, 6.0],],
+                        [[40.0, 1.5, 2.5], [50.0, 3.5, 4.5], [0.0, 0.0, 0.0],],
+                    ],
+                    dtype=np.float32,
+                ),
+                [[[1.0], [1.0], [1.0]], [[1.0], [1.0], [0.0]]],
+                realistic_feature_dense_seq_1.numpy(),
+            ),
+        ),
+    ],
+)
+def test_sequence_layer_correct_output(attribute_signature, inputs, expected_outputs):
+    layer = RasaSequenceLayer(
+        attribute=attribute_name,
+        config=dict(
+            model_config_transformer_mlm, **{HIDDEN_LAYERS_SIZES: {attribute_name: []}}
+        ),
+        attribute_signature=attribute_signature,
+    )
+
+    (
+        seq_sent_features_expected,
+        mask_seq_sent_expected,
+        token_ids_expected,
+    ) = expected_outputs
+
+    _, seq_sent_features, mask_seq_sent, token_ids, mlm_boolean_mask, _ = layer(
+        inputs, training=True
```

Comment:
could you please also check that mlm is not applied if `training=False`

Reply:
Sure, I've now added a check. Though it's just an indirect one -- checking that no `token_ids` or `mlm_boolean_mask` are returned at test time.

Reply:
you can check that transformed text is exactly as it is expected

Reply:
Ah, you're right. If I deactivate the hidden layers and the transformer, I can actually check it. Thanks!

Reply:
This should now be fixed.